### PR TITLE
Create a moderated view for the comment

### LINKED ISSUE #4048: Create a moderated view for the comment
**This issue is a part of Planning issue #3331**

Currently, we only moderate nodes before publishing, now we are planning for the comment moderation system. 

Show the comment on dashboard to moderators and admins in moderated view if the comment has status = 4. The design of the moderated comment can resemble that of moderated node.

The views which would require changes like in `app/views/dashboard`. the moderated node view lies in the same folder. 

**Hi, this issue has been opened as a task on GCI dashboard. If you are Google Code-In participant and want to solve this issue, please claim it from Google Code-In [website](https://codein.withgoogle.com/organizations/public-lab/) . Thanks!**

### PR DESCRIPTION
**Pull request by ValentinaTironi**
Fixes #4048 

screen shot 2018-12-09 at 5 52 32 pm

### CODE REVIEW

**Comment by ValentinaTironi** [comment]
Hey @gauravano @jywarren check out this :) The errors are from some test, It comes from master branch

**Comment by ValentinaTironi** [comment]
@SidharthBansal

**Comment by ValentinaTironi** [comment]
@gauravano Here is my pull request to the view for the comment

**Comment by gauravano** [comment]
Have you claimed this task? @ValentinaTironi

**Comment by ValentinaTironi** [comment]
noup, I'm waiting for approval of that https://github.com/publiclab/plots2/pull/4217

**Comment by ValentinaTironi** [comment]
Now yes, I've claimed this @gauravano :)

**Comment by gauravano** [comment]
@ValentinaTironi it's approved on GCI dash

**Comment by ValentinaTironi** [comment]
OMG, that was so fast! Thank you! I already claimed another one hahaha

**Comment by SidharthBansal** [comment]
I approved it. Please take another task immediately on the GCI dashboard

**Comment by ValentinaTironi** [comment]
Ready this one too https://codein.withgoogle.com/dashboard/task-instances/6570167489265664/
Here is the pull request https://github.com/publiclab/plots2/pull/4226 💯

**Comment by ValentinaTironi** [comment]
@jywarren Also you can merge this one :)

**Comment by jywarren** [comment]
would you be able to make the buttons appear on the same line as the message, to save a little space? Thank you!!!

**Comment by ValentinaTironi** [comment]
Yeah, I'm gonna have lunch, and I'll do that :)

**Comment by gauravano** [comment]
I am pushing this to unstable for testing. Thanks!

**Comment by ValentinaTironi** [comment]
What do you think @jywarren? :)
screen shot 2018-12-10 at 7 34 18 pm

**Comment by jywarren** [comment]
Perfect!

**Comment by jywarren** [comment]
If it looks good to you Gaurav go ahead and merge it! Thanks!!

File: app/views/dashboard/_comment_moderate.html.erb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<% if comment.status == 4 %>
+  <p class="alert alert-warning moderated">
+    <% if logged_in_as(['admin','moderator']) %>
+      <%= t('dashboard._comment_moderate.first_time_post') %>
+      <a class="btn btn-default btn-xs" href="/admin/publish_comment/<%= comment.id %>"><%= t('dashboard.moderate.approve') %></a>
+      <a class="btn btn-default btn-xs" href="/admin/mark_spam/<%= comment.id %>"><%= t('dashboard.moderate.spam') %></a>
```
Comment (gauravano) [inline app/views/dashboard/_comment_moderate.html.erb]: @ValentinaTironi this should be `mark_comment_spam` as `mark_spam` is used for marking node as spam.

Comment (ValentinaTironi) [reply app/views/dashboard/_comment_moderate.html.erb]: Got it! I'm gonna change it right now!

**Comment by ValentinaTironi** [comment]
Done @gauravano!

File: app/views/dashboard/_comment_moderate.html.erb (hunk 1 above)
Comment (gauravano) [reply app/views/dashboard/_comment_moderate.html.erb]: Also, you could test your work at https://unstable.publiclab.org . Thanks!

**Comment by ValentinaTironi** [comment]
Let me know if I can help in something else. Now, I'll sleep a few hours, I've headache and If I'm looking the computer more time my eyes are gonna headache hahaha Thank you so much! See you later :)

**Comment by gauravano** [comment]
Don't worry about PR @ValentinaTironi, a good sleep is indeed important. 

> If I'm looking the computer more time my eyes are gonna blow up hahaha

I can feel you. Good Night!

File: app/controllers/admin_controller.rb [hunk 2]
```diff
@@ -139,7 +139,7 @@ def mark_spam
   def mark_comment_spam
     @comment = Comment.find params[:id]
     if current_user && (current_user.role == 'moderator' || current_user.role == 'admin')
-      if @comment.status == 1
+      if @comment.status == 1 || @comment.status == 4
```
Comment (gauravano) [inline app/controllers/admin_controller.rb]: Change in mark_comment_spam so that moderated comment can be marked spam.

**Comment by gauravano** [comment]
Merged :tada: :100: !

File: app/controllers/admin_controller.rb (hunk 2 above)
Comment (ValentinaTironi) [reply app/controllers/admin_controller.rb]: Got it. Do I have to create a new pull request for that?

**Comment by jywarren** [comment]
Awesome!!!

**Comment by gauravano** [comment]
Issue to be fixed #4447.

**Pull request merged**